packages/contracts-bedrock/test/dispute/zk/OPSuccinctFaultDisputeGame.t.sol

```diff
@@ -0,0 +1,787 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+// Testing
+import "forge-std/Test.sol";
+
+// Libraries
+import { Claim, Duration, GameStatus, GameType, Hash, Timestamp, Proposal } from "src/dispute/lib/Types.sol";
+import {
+    BadAuth,
+    IncorrectBondAmount,
+    UnexpectedRootClaim,
+    NoCreditToClaim,
+    GameNotFinalized,
+    ParentGameNotResolved,
+    InvalidParentGame,
+    ClaimAlreadyChallenged,
+    GameOver,
+    GameNotOver,
+    IncorrectDisputeGameFactory
+} from "src/dispute/lib/Errors.sol";
+import { OP_SUCCINCT_FAULT_DISPUTE_GAME_TYPE } from "src/dispute/lib/Types.sol";
+import { Constants } from "src/libraries/Constants.sol";
+
+// Contracts
+import { DisputeGameFactory } from "src/dispute/DisputeGameFactory.sol";
+import { OPSuccinctFaultDisputeGame } from "src/dispute/zk/OPSuccinctFaultDisputeGame.sol";
+import { AnchorStateRegistry } from "src/dispute/AnchorStateRegistry.sol";
+import { SystemConfig } from "src/L1/SystemConfig.sol";
+import { SuperchainConfig } from "src/L1/SuperchainConfig.sol";
+import { AccessManager } from "src/dispute/zk/AccessManager.sol";
+import { SP1MockVerifier } from "./mocks/SP1MockVerifier.sol";
+
+// Interfaces
+import { IDisputeGame } from "interfaces/dispute/IDisputeGame.sol";
+import { IDisputeGameFactory } from "interfaces/dispute/IDisputeGameFactory.sol";
+import { ISP1Verifier } from "src/dispute/zk/ISP1Verifier.sol";
+import { ISuperchainConfig } from "interfaces/L1/ISuperchainConfig.sol";
+import { ISystemConfig } from "interfaces/L1/ISystemConfig.sol";
+import { IAnchorStateRegistry } from "interfaces/dispute/IAnchorStateRegistry.sol";
+import { Proxy } from "src/universal/Proxy.sol";
+
+/// @title OPSuccinctFaultDisputeGame_TestInit
+/// @notice Base test contract with shared setup for OPSuccinctFaultDisputeGame tests.
+abstract contract OPSuccinctFaultDisputeGame_TestInit is Test {
```

Comment:
It would be cleaner for this contract to reuse the test facility provided by [`DIsputeGameFactory_TestInit`](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/test/dispute/DisputeGameFactory.t.sol#L51). That base contract sets up op contracts and makes it easy to add new game implementations to the factory. Using the base contract will drastically simplify the test setup done here.